app/src/main/java/com/github/onlynotesswent/ui/common/NoteComposables.kt

```diff
@@ -168,6 +163,100 @@ fun NoteItem(
         }
       }
 }
+
+/**
+ * Displays a bottom sheet with options to move or delete a note. The bottom sheet is displayed when
+ * the user clicks on the more options icon in the note card.
+ *
+ * @param note The note data that will be displayed in this card.
+ * @param noteViewModel The ViewModel that provides the list of notes to display.
+ * @param folderViewModel the folderViewModel used here to move the note.
+ * @param navigationActions The navigation instance used to transition between different screens.
+ * @param onDismiss The callback to be invoked when the bottom sheet is dismissed.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun NoteOptionsBottomSheet(
+    note: Note,
+    noteViewModel: NoteViewModel,
+    folderViewModel: FolderViewModel,
+    navigationActions: NavigationActions,
+    onDismiss: () -> Unit
+) {
+  var showFileSystemPopup by remember { mutableStateOf(false) } // State to show FileSystemPopup
+  var showDeletePopup by remember { mutableStateOf(false) }
+
+  if (showFileSystemPopup) {
+    FileSystemPopup(
+        onDismiss = { showFileSystemPopup = false },
+        folderViewModel = folderViewModel,
+        onMoveHere = { selectedFolder ->
+          noteViewModel.updateNote(note.copy(folderId = selectedFolder?.id))
+          showFileSystemPopup = false
+          // this is needed to recompose the screen
+          noteViewModel.getNotesFromFolder(folderViewModel.selectedFolder.value?.id ?: "")
+          folderViewModel.selectedFolder.value?.let { folderViewModel.getFolderById(it.id) }
+          onDismiss() // Dismiss the bottom sheet after moving the note
+        })
+  }
+
+  if (showDeletePopup) {
+    ConfirmationPopup(
+        title = stringResource(R.string.delete_note),
+        text = stringResource(R.string.delete_note_text),
+        onConfirm = {
+          noteViewModel.deleteNoteById(note.id, note.userId)
+          navigationActions.navigateTo(TopLevelDestinations.OVERVIEW) // Navigate to overview
```

Comment:
Why do we navigate to Overview, can't we stay in the same folder?

Reply:
I copied what should be done with what was previously done with the edit note in the edit note screen. I agree with you that we should not go back to overview and therefore deleted this call.